common/test/concurrent/FutureSemaphoreTest.scala

```diff
@@ -0,0 +1,37 @@
+package concurrent
+
+import akka.actor.ActorSystem
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FlatSpec, Matchers}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class FutureSemaphoreTest extends FlatSpec with Matchers with ScalaFutures {
```

Comment:
This is horrible to test since it requires blocking to test non-deterministic behaviour. Happy to remove, what do you think?

Reply:
I think that's fine really. It adds a couple of 2 second delays but that's not a big deal I think and definitely need to test this kind of behaviour to make sure it works.